# Data Analytics on Superstore dataset

### We are going to perform analytics on superstore datasets.

### Import the required libraries

In [26]:
# Import the numerical Python 
# Numpy is used for performing numerical operations on the different data structures 
import numpy as np

# Import the pandas library
# Pandas is used for converting different data structures into pandas dataframe
import pandas as pd

# Import the seaborn
# Seaborn is used to visualize the data structures
import seaborn as sns

# Import the matplotlib
# Matplotlib is the major library utilized for visualization of any data structure
import matplotlib.pyplot as plt

### Load the dataset

In [29]:
# Read the dataset from local
df_superstore = pd.read_excel('C://Users//Asus//Desktop//superstore.xlsx')

### Data overview

In [34]:
# Show the first 5 observation(row) of the data
df_superstore.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

In [38]:
# Show me the last 5 observation
df_superstore.tail()

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
9989    9990  CA-2014-110422 2014-01-21 2014-01-23    Second Class   
9990    9991  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9991    9992  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9992    9993  CA-2017-121258 2017-02-26 2017-03-03  Standard Class   
9993    9994  CA-2017-119914 2017-05-04 2017-05-09    Second Class   

     Customer ID     Customer Name   Segment        Country         City  ...  \
9989    TB-21400  Tom Boeckenhauer  Consumer  United States        Miami  ...   
9990    DB-13060       Dave Brooks  Consumer  United States   Costa Mesa  ...   
9991    DB-13060       Dave Brooks  Consumer  United States   Costa Mesa  ...   
9992    DB-13060       Dave Brooks  Consumer  United States   Costa Mesa  ...   
9993    CC-12220      Chris Cortes  Consumer  United States  Westminster  ...   

     Postal Code  Region       Product ID         Category Sub-Category  \
9989       33180   South  FUR-FU-10001889        Furniture  Furnishings   
9990       92627    West  FUR-FU-10000747        Furniture  Furnishings   
9991       92627    West  TEC-PH-10003645       Technology       Phones   
9992       92627    West  OFF-PA-10004041  Office Supplies        Paper   
9993       92683    West  OFF-AP-10002684  Office Supplies   Appliances   

                                           Product Name    Sales  Quantity  \
9989                             Ultra Door Pull Handle   25.248         3   
9990  Tenex B1-RE Series Chair Mats for Low Pile Car...   91.960         2   
9991                              Aastra 57i VoIP phone  258.576         2   
9992  It's Hot Message Books with Stickers, 2 3/4" x 5"   29.600         4   
9993  Acco 7-Outlet Masterpiece Power Center, Wihtou...  243.160         2   

      Discount   Profit  
9989       0.2   4.1028  
9990       0.0  15.6332  
9991       0.2  19.3932  
9992       0.0  13.3200  
9993       0.0  72.9480  

[5 rows x 21 columns]

### Check the shape of the data

In [41]:
df_superstore.shape

(9994, 21)

**Interpretation**
- We have 9994 observations(rows) in the data
- We have 21 attributes(columns) in the data 

In [44]:
# Derive the basic information 
df_superstore.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9994 non-null   int64         
 1   Order ID       9994 non-null   object        
 2   Order Date     9994 non-null   datetime64[ns]
 3   Ship Date      9994 non-null   datetime64[ns]
 4   Ship Mode      9994 non-null   object        
 5   Customer ID    9994 non-null   object        
 6   Customer Name  9994 non-null   object        
 7   Segment        9994 non-null   object        
 8   Country        9994 non-null   object        
 9   City           9994 non-null   object        
 10  State          9994 non-null   object        
 11  Postal Code    9994 non-null   int64         
 12  Region         9994 non-null   object        
 13  Product ID     9994 non-null   object        
 14  Category       9994 non-null   object        
 15  Sub-Category   9994 n

**Interpretation**
- In the data, we have 6 numerical columns(float + int)
- In the data, we have 13 categorical columns(object)
- Memory usage is 1.6+ MB
- In the data, we dont have null values

In [47]:
# Describe the dataset
df_superstore.describe().T

count                           mean                  min  \
Row ID       9994.0                         4997.5                  1.0   
Order Date     9994  2016-04-30 00:07:12.259355648  2014-01-03 00:00:00   
Ship Date      9994  2016-05-03 23:06:58.571142912  2014-01-07 00:00:00   
Postal Code  9994.0                   55190.379428               1040.0   
Sales        9994.0                     229.858001                0.444   
Quantity     9994.0                       3.789574                  1.0   
Discount     9994.0                       0.156203                  0.0   
Profit       9994.0                      28.656896            -6599.978   

                             25%                  50%                  75%  \
Row ID                   2499.25               4997.5              7495.75   
Order Date   2015-05-23 00:00:00  2016-06-26 00:00:00  2017-05-14 00:00:00   
Ship Date    2015-05-27 00:00:00  2016-06-29 00:00:00  2017-05-18 00:00:00   
Postal Code              23223.0              56430.5              90008.0   
Sales                      17.28                54.49               209.94   
Quantity                     2.0                  3.0                  5.0   
Discount                     0.0                  0.2                  0.2   
Profit                   1.72875               8.6665               29.364   

                             max          std  
Row ID                    9994.0  2885.163629  
Order Date   2017-12-30 00:00:00          NaN  
Ship Date    2018-01-05 00:00:00          NaN  
Postal Code              99301.0  32063.69335  
Sales                   22638.48   623.245101  
Quantity                    14.0      2.22511  
Discount                     0.8     0.206452  
Profit                  8399.976   234.260108

**Interpretation**
- Average Row ID is 4997.5,Minimum Row ID is 1.0,Maximum Row ID is 9994.0
- Average Order Date is 2016-04-30 00:07:12.259355648,Minimum Order Date is 2014-01-03 00:00:00,Maximum Order Date is 2017-12-30 00:00:00.
- Average Ship Date is 2016-05-03 23:06:58.571142912 ,Minimum  Ship Date is 2014-01-07 00:00:00,Maximum Ship Date is 2018-01-05 00:00:00.
- Average Postal code is 55190.379428, Minimum Postal code is 1040.0, Maximum postal code is 99301.0.
- Average Sales is 229.858001, Minimum Sales is 0.444, Maximum sales is 22638.48.
- Average Quantity is 3.789574, Minimum Quantity is 1.0, Maximum Quantity is 14.0
- Average Discount is 0.156203, Minimum Discount is 0.0, Maximum Discount is 0.8.
- Average Profit is 28.656896, Minimum Profit is 6599.978, Maximum Profit is 234.260108